iceoryx_examples/icedelivery/iox_subscriber_untyped_modern.cpp

```diff
@@ -35,25 +35,22 @@ static void sigHandler(int f_sig [[gnu::unused]])
 void subscriberHandler(iox::popo::WaitSet& waitSet)
 {
     // run until interrupted
-    while(!killswitch)
+    while (!killswitch)
     {
         auto triggeredConditions = waitSet.wait();
-        for(auto& condition : triggeredConditions)
+        for (auto& condition : triggeredConditions)
         {
             auto untypedSubscriber = dynamic_cast<iox::popo::UntypedSubscriber*>(condition);
-            if(untypedSubscriber)
+            if (untypedSubscriber)
             {
                 untypedSubscriber->take()
-                        .and_then([](iox::cxx::optional<iox::popo::Sample<const void>>& allocation){
-                                auto position = reinterpret_cast<const Position*>(allocation->get());
-                                std::cout << "Got value: (" << position->x << ", " << position->y << ", " << position->z << ")" << std::endl;
-                        })
-                        .if_empty([]{
-                            std::cout << "Didn't get a value, but do something anyway." << std::endl;
-                        })
-                        .or_else([](iox::popo::ChunkReceiveError){
-                            std::cout << "Error receiving chunk." << std::endl;
-                        });
+                    .and_then([](iox::cxx::optional<iox::popo::Sample<const void>>& allocation) {
```

Comment:
Could we use `auto & allocation` here?

Reply:
Try it 😉 Type deduction is very limited until C++17